pkg/txm/txm.go

```diff
@@ -53,6 +53,7 @@ type Request struct {
 	Amount          tlb.Coins       // Amount in nanotons
 	Bounce          bool            // Bounce on error (TON message flag)
 	StateInit       *cell.Cell      // Optional: contract deploy init
+	IdempotencyKey  *string         // Optional: unique ID for transaction tracking (similar to EVM pattern)
```

Comment:
The name is misleading, the point of an idempotency key is to detect semantically identical transactions that are already in flight (and replace the old transaction currently in the queue/etc to avoid duplicate writes). That's what I assumed but looking at the implementation it's only used for logging.

If we just want an identifier then ID/TransactionID seems a better fit. Solana calls this ID: https://github.com/smartcontractkit/chainlink-solana/blob/aaac72331fc8e3448a7b3695c6d29df8c0716d46/pkg/solana/txm/txm.go#L719-L723